{
  try {
    action(e);
  }
 catch (  Throwable t) {
    ExceptionReporter.showDialog(t);
  }
}
